tests/conftest.py

```
@@ -89,66 +114,121 @@ def _get_jwt_auth_headers(server_url: str, user_name: str | None = None) -> dict
-        headers=auth_headers,
-    )
+    sio = socketio.Client()
+    sio.connect(server, auth={"token": jwt_token}, wait=True)
 
-    # If room doesn't exist (404), create it then join
-    if response.status_code == 404:
+    response = sio.call("room:join", {"roomId": room_id, "clientType": "frontend"})
+
+    # Handle room not found - create it first
+    if response.get("code") == 404:
         create_response = requests.post(
```

Ruff (0.14.10)

Probable use of `requests` call without timeout

(S113)